### PR TITLE
New chat should automatically inherit context of the previous one

### LINKED ISSUE #596: [UX QA] Remove actions to store and restore default context
As it sits today, these two actions are a bit confusing.

Even though we will need this functionality in the future, presenting them as we do today creates more harm than good.

So my recommendation is to remove these actions until we have a better flow:

Image

### PR DESCRIPTION
**Pull request by pkukielka**
Fixes https://github.com/sourcegraph/jetbrains/issues/596

## Changes

* `Save as Default Context`, and `Restore Default Context` buttons removed
* Opening new chat inherits context settings from the previously opened one
* Existing chats preserves their context (that was agreed with @danielmarquespt )

## Test plan

As in updated TESTING.MD:

### Free/pro accounts:

1. Open `sourcegraph/cody` project with non-enterprise account.
2. Open new chat and ask question about current repo (e.g. some class) - assistant should know the answer.
3. Open new chat and ask question about squirrel - assistant should describe you an animal.
4. Open new chat and disable local context. Ask about current repo (e.g. some class) - assistant should not have a
   context.
5. Save current context as default. Close the IDE. Reopen the IDE.
    - Go to Chat History tab and open previous chats one by one. Both history and context settings are properly
      preserved.
    - Open new chat. Context should be disabled, as we previously set that as new default. Enable it again and set as
      default.

### Enterprise accounts:

1. Open `sourcegraph/cody` project with enterprise account.
2. Re-do all check from `Testing free/pro accounts` section but now with enterprise account.
3. Click [+] button in the context panel and type sourcegraph repo url (`github.com/sourcegraph/sourcegraph`)
    - Validator should block accepting incomplete or invalid URL.
    - Add the `sourcegraph/sourcegraph` repo by hitting Add button.
4. Open new chat and ask question about squirrel - assistant should describe you an HTTP server, **NOT** animal.
5. Open new chat and disable `sourcegraph/sourcegraph` remote repo context.
7. Ask question about squirrel. It should again describe you an animal or have no context.
8. Close the IDE. Reopen the IDE.
    - Go to Chat History tab and open previous chats one by one. Check if both history and context settings are properly
      preserved.
    - Open new chat and check if it properly inherits all setting from previously opened historical chat
    - If `sourcegraph/sourcegraph` repo was previously added please remove it by selecting it, and then clicking [-]
      button.
    - Ask question about squirrel. It should again describe you an animal or have no context.

### CODE REVIEW

File: src/main/resources/CodyBundle.properties [hunk 1]
```diff
@@ -76,7 +76,7 @@ context-panel.button.restore-default=Restore Default Context
 context-panel.button.help=Help
 context-panel.in-progress=Running Cody 'Keyword Search' indexer...
 context-panel.tree.root=repositories-root
-context-panel.tree.node-chat-context=Chat Context
+context-panel.tree.node-chat-context=Context for this Chat
```
Comment (mkondratek) [inline src/main/resources/CodyBundle.properties:79]: 👍

**Review by mkondratek — APPROVED** [review]

**Comment by mkondratek** [comment]
I sent a few comments via Slack. Nothing blocking.

File: src/main/resources/CodyBundle.properties [hunk 2]
```diff
@@ -76,7 +76,7 @@ context-panel.button.restore-default=Restore Default Context
 context-panel.button.help=Help
```
Comment (exigow) [inline src/main/resources/CodyBundle.properties:76]: We should remove unused button labels (save-default and remove-default).

File: src/main/kotlin/com/sourcegraph/cody/chat/CodyChatMessageHistory.kt [hunk 3]
```diff
@@ -59,7 +59,7 @@ class CodyChatMessageHistory(
 
   private fun preloadHistoricalMessages(chatSession: ChatSession) {
     HistoryService.getInstance(project)
-        .getHistoryReadOnly()
+        .state
```
Comment (exigow) [inline src/main/kotlin/com/sourcegraph/cody/chat/CodyChatMessageHistory.kt:62]: Looks unrelated. Are we fixing something here?

**Review by exigow — APPROVED** [review]
Looks good :+1:, added one minor comment.
